lib/syntax.ml

```diff
@@ -255,9 +255,27 @@ let rec resolve_columns env expr =
       let schema = Schema.Source.from_schema schema in
       (* represet nested selects as functions with sql parameters as function arguments, some hack *)
       match schema, usage with
-      | [ {domain;_} ], `AsValue -> ResFun (Type.Ret domain.t, as_params p) (* use nullable *)
+      | [ {domain;_} ], `AsValue ->
+        (* This function should be raised? *)
+        let rec with_count = function 
+          | Fun (Agg Count, _)
+          | SelectExpr (_, _) -> Some domain
+          | Fun (_, exprs) -> List.find_map with_count exprs
+          | Value _| Param _| Inparam _| Choices (_, _) | InChoice (_, _, _)
+          | Column _| Inserted _| InTupleList (_, _) -> None
+        in
+        let default_null = Type.make_nullable domain in
+        (* The only way to have a result in a subquery is to use the count function. 
+           Any other expression could possibly return no rows. *)
+        let typ = match select.select_complete.select with 
+        | ({ having = Some _; _ }, _) -> Type.nullable domain.t
+        | ({ columns = [Expr(c, _)]; _ }, _) -> c |> with_count |> Option.default default_null
+        | ({ columns = [_]; _ }, _) -> default_null
+        | _ -> raise (Schema.Error (schema, "Different count of selected columns are presented"))
```

Comment:
this error wouldn't make sense to me without looking at the code. Should say smth like "nested sub-select used as an expression returns more than one column"

Reply:
BTW I think it's possible in MySQL 8 to return the row result.